service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/ManageOrdersEventBuilderTest.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.fpl.service.orders;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.fpl.events.GeneratedOrderEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.order.generated.GeneratedOrder;
+import uk.gov.hmcts.reform.fpl.service.orders.history.SealedOrderHistoryService;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+class ManageOrdersEventBuilderTest {
+    private final DocumentReference document = mock(DocumentReference.class);
+    private final GeneratedOrder order = mock(GeneratedOrder.class);
+    private final CaseData caseData = mock(CaseData.class);
+    private final CaseData caseDataBefore = mock(CaseData.class);
+    private final List<Element<GeneratedOrder>> orders = wrapElements(order);
+
+    private final SealedOrderHistoryService historyService = mock(SealedOrderHistoryService.class);
+    private final ManageOrdersEventBuilder underTest = new ManageOrdersEventBuilder(historyService);
+
+    @Test
+    void buildAmended() {
+        when(caseData.getOrderCollection()).thenReturn(orders);
+        when(caseDataBefore.getOrderCollection()).thenReturn(orders);
+
+        Optional<GeneratedOrderEvent> event = underTest.build(caseData, caseDataBefore);
+
+        assertThat(event).isEmpty();
+    }
+
+    @Test
+    void buildNonAmended() {
+        List<Element<GeneratedOrder>> ordersBefore = List.of();
+        when(caseData.getOrderCollection()).thenReturn(orders);
+        when(caseDataBefore.getOrderCollection()).thenReturn(ordersBefore);
+        when(historyService.lastGeneratedOrder(caseData)).thenReturn(order);
+        when(order.getDocument()).thenReturn(document);
+
+        Optional<GeneratedOrderEvent> event = underTest.build(caseData, caseDataBefore);
+
+        assertThat(event).isPresent();
```

Comment:
assertThat(event).isEqalTo(Optional.of()) ?

Reply:
`GeneratedOrderEvent` doesn't overwrite its equals method, as such it only compares by reference instead of by fields.  Therefore wrapping an expected event in an optional will not work, so I assert that the optional is populated then assert the specific fields.